### ISSUE TITLE
Drop support for Python 3.6

### ISSUE THREAD

**Issue by edgarrmondragon** [transferwise/pipelinewise-singer-python#63]
Some dependencies of this package, like `backoff` (see https://github.com/transferwise/pipelinewise-singer-python/pull/61), can not currently be updated because they already dropped support for Python 3.6.

Python 3.6 EOL was back on 2021/12/23, and [pypistats.org](https://pypistats.org/packages/pipelinewise-singer-python) shows less than one percent of installations are on 3.6:

<img width="1457" alt="Screen Shot 2022-05-03 at 14 14 31" src="https://user-images.githubusercontent.com/16805946/166526503-fc4b6518-5b7e-4b17-a44c-4d95a8aede4f.png">

**Comment by astrojuanlu** [comment]
Is something like [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html) feasible to adopt? (3.8+ at the moment, 3.9+ one year from now). With the cadence of Python versions, supporting 3.6, 3.7, 3.8, 3.9, and 3.10 feels a bit of a stretch.

Alternatively, something like "the default version installed in the oldest Ubuntu LTS", which is [what Sphinx does](https://www.sphinx-doc.org/en/master/internals/release-process.html#python-version-support-policy). This again means 3.8+ as of now.

In other words, not only I support dropping 3.6, I also support skipping 3.7 altogether... and then apply something like [pyupgrade](https://github.com/asottile/pyupgrade).

**Comment by Samira-El** [comment]
Hey Edgar, we no longer use python 3.6 so this is fine by me.
  
We've been dropping 3.6 from taps/targets we maintain, but as you might have noticed, there are too many repos so it's a slow process 😅 and we have some more internal projects this quarter. So feel free to submit a PR to drop support for 3.6 :)  


Juan, we've just recently moved to Python 3.8 as our main python version for pipelinewise and all its connectors, it's our first python upgrade so we don't have a strategy set in stone yet, we decided on 3.8 because it's the default version of latest ubuntu LTS and we already had CI changes against it so it was easier to upgrade without any breaking changes or code changes. 
We kept 3.7 because it was in the CI matrix and it was no harm to keep it in case they are people using it.

I think we're open to exploring and having a proper upgrade strategy, but we don't currently have capacity for this.

**Comment by edgarrmondragon** [comment]
Hi @Samira-El. Thanks for your response. Yeah, I'll start work on a PR in the next days 🙂
